tedana/model/combine.py

```diff
@@ -45,45 +128,47 @@ def make_optcom(data, tes, mask, t2s=None, combmode='t2s', verbose=True):
     2.  Perform weighted average per voxel and TR across TEs based on weights
         estimated in the previous step.
     """
-    if combmode == 't2s' and t2s is None:
+    if data.ndim != 3:
+        raise ValueError('Input data must be 3D (S x E x T)')
+
+    if len(tes) != data.shape[1]:
+        raise ValueError('Number of echos provided does not match second '
+                         'dimension of input data: {0} != '
+                         '{1}'.format(len(tes), data.shape[1]))
+
+    if mask.ndim != 1:
+        raise ValueError('Mask is not 1D')
+    elif mask.shape[0] != data.shape[0]:
+        raise ValueError('Mask and data do not have same number of '
+                         'voxels/samples: {0} != {1}'.format(mask.shape[0],
+                                                             data.shape[0]))
+
+    if combmode not in ['t2s', 'ste']:
+        raise ValueError("Argument 'combmode' must be either 't2s' or 'ste'")
+    elif combmode == 't2s' and t2s is None:
         raise ValueError("Argument 't2s' must be supplied if 'combmode' is "
                          "set to 't2s'.")
     elif combmode == 'ste' and t2s is not None:
         LGR.warning("Argument 't2s' is not required if 'combmode' is 'ste'.")
 
-    _, _, n_vols = data.shape
     mdata = data[mask, :, :]  # mask out empty voxels/samples
     tes = np.array(tes)[np.newaxis]  # (1 x E) array_like
 
     if t2s.ndim == 1:
         msg = 'Optimally combining data with voxel-wise T2 estimates'
         ft2s = t2s[mask, np.newaxis]
     else:
-        msg = 'Optimally combining data with voxel- and volume-wise T2 estimates'
+        msg = ('Optimally combining data with voxel- and volume-wise T2 '
+               'estimates')
         ft2s = t2s[mask, :, np.newaxis]
 
     if verbose:
         LGR.info(msg)
 
     if combmode == 'ste':
-        alpha = mdata.mean(axis=-1) * tes
-        alpha = np.tile(alpha[:, :, np.newaxis], (1, 1, n_vols))
+        combined = _combine_ste(mdata, tes)
```

Comment:
I love that we're splitting these up, here ! Just to confirm, do we now have unit test for each option, too ?

Reply:
I've added tests for these two and for `make_optcom`, but they just check the output array shapes. We'll want to add ones that use valid input data and check the actual results at some point, I'm sure.